### DISCUSSION TITLE
Lesson 7: how to get the second element of an array using storage ?

### DISCUSSION THREAD

**Discussion by devmousa** [Cyfrin/foundry-full-course-cu#695] [Q&A] [answered]
in the ```Storage``` section in the lesson 7, I take the script in the file ```DeployStorageFun.s.sol``` and it works very well

this function returns the first element of an array:
```solidity
    function printFirstArrayElement(address contractAddress) public view {
        bytes32 arrayStorageSlotLength = bytes32(uint256(2));
        bytes32 firstElementStorageSlot = keccak256(abi.encode(arrayStorageSlotLength));
        bytes32 value = vm.load(contractAddress, firstElementStorageSlot);
        console.log("First element in array:");
        console.logBytes32(value);
    }
```

I'm wondering, how to get the second element of the array ?

*Note*: it's already returning the first element, but when I added another element, it's still returning the first element, so I want to know how can I get the second or the third element using these storage functions ?

Thank you all in advance and happy coding ;)

**Comment by shikhar229169** [comment, marked as answer]
```solidity
bytes32 secondElementStorageSlot = bytes32(uint256(firstElementStorageSlot) + 1);
bytes32 thirdElementStorageSlot = bytes32(uint256(firstElementStorageSlot) + 2);
bytes32 kthElementStorageSlot = bytes32(uint256(firstElementStorageSlot) + k - 1);
```

**Reply by devmousa** [reply]
Thank you a lot 😊
